operations/src/main/java/no/sikt/nva/handle/HandleDatabase.java

```diff
@@ -0,0 +1,140 @@
+package no.sikt.nva.handle;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+import no.sikt.nva.handle.exceptions.CreateHandleException;
+import nva.commons.core.Environment;
+import nva.commons.core.JacocoGenerated;
+import nva.commons.core.paths.UriWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+@JacocoGenerated
+public class HandleDatabase {
+
+    private static final Logger logger = LoggerFactory.getLogger(HandleDatabase.class);
+    public static final String CREATE_ID_SQL = "INSERT INTO handle(date_created) VALUES ( current_timestamp ) "
+                                               + "RETURNING handle_id";
+    public static final String SET_HANDLE_SQL = "UPDATE handle SET handle =  ?, url = ? WHERE handle_id = ?";
+    public static final String SET_URI_SQL = "UPDATE handle SET url = ? WHERE handle = ?";
+    public static final String CHECK_URL_SQL = "SELECT handle FROM handle WHERE url = ?";
+    public static final String REUSED_EXISTING_HANDLE_FOR_URI = "Reused existing handle '%s' for URI '%s'";
+    public static final String CREATED_HANDLE_FOR_URI = "Created handle '%s' for URI '%s'";
+    public static final String ERROR_CREATING_HANDLE_FOR_URI = "Error creating handle for URI '%s'";
+    public static final String ERROR_UPDATING_HANDLE_FOR_URI = "Error updating handle '%s/%s'";
+    public static final String CHARACTER_SLASH = "/";
+    public static final String ENV_HANDLE_PREFIX = "HANDLE_PREFIX";
+    public static final String ENV_HANDLE_BASE_URI = "HANDLE_BASE_URI";
+    public static final int ONE_ROW = 1;
+
+    private final Environment environment;
+
+    public HandleDatabase(Environment environment) {
+        this.environment = environment;
+    }
+
+    public URI createHandle(URI uri, Connection connection) throws CreateHandleException, SQLException {
+        try {
+            URI existingHandle = fetchExistingHandleByValue(uri, connection);
+            if (nonNull(existingHandle)) {
+                logger.info(String.format(REUSED_EXISTING_HANDLE_FOR_URI, existingHandle, uri));
+                return existingHandle;
+            } else {
+                return createNewHandle(uri, connection);
+            }
+        } catch (SQLException e) {
+            var message = String.format(ERROR_CREATING_HANDLE_FOR_URI, uri) + ": " + e.getMessage();
+            logger.error(message, e);
+            throw new CreateHandleException(message);
```

Comment:
`createHandle` is throwing `CreateHandleException`, while `updateHandle` is throwing `SQLException`.
`CreateHandleException` extends `ApiGatewayException`, I would say that the database layer should not throw ApiGatewayExceptions?